plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -841,7 +841,15 @@ def _write_pulp_distribution_file(self, distro_files, old_xml_file_path):
             element.text = distro_file
 
         tree = cElementTree.ElementTree(new_xml_root)
-        tree.write(os.path.join(self.get_working_dir(), constants.DISTRIBUTION_XML))
+        distribution_xml_path = os.path.join(self.get_working_dir(), constants.DISTRIBUTION_XML)
+        try:
+            os.remove(distribution_xml_path)
+        except OSError, e:
```

Comment:
It's nice to use ``except OSError as e`` since that's compatible with Python 3 - It's not exactly difficult to automatically fix this, but it would make a future diff slightly smaller :smile: